doc/src/devdocs/locks.md

```diff
@@ -3,158 +3,137 @@
 The following strategies are used to ensure that the code is dead-lock free (generally by addressing
 the 4th Coffman condition: circular wait).
 
-> 1. structure code such that only one lock will need to be acquired at a time
-> 2. always acquire shared locks in the same order, as given by the table below
-> 3. avoid constructs that expect to need unrestricted recursion
+1. structure code such that only one lock will need to be acquired at a time
+2. always acquire shared locks in the same order, as given by the table below
+3. avoid constructs that expect to need unrestricted recursion
+
+## Types of locks
+
+`uv_mutex_t` (or `std::mutex`) is a wrapper around platform-specific locks
+(`pthread_mutex_t` on Unix, `CRITICAL_SECTION` on Windows).  It may cause the
```

Comment:
```suggestion
(`pthread_mutex_t` on Unix, `SRWLock` on Windows).  It may cause the
```
I don't think it has been a CRITICAL_SECTION in several years as they are much much slower

Reply:
https://github.com/libuv/libuv/blob/master/include/uv/win.h. I guess that is right. We may want to change that upstream